src/main/java/com/pnudev/librarysystem/service/BookService.java

```diff
@@ -0,0 +1,141 @@
+package com.pnudev.librarysystem.service;
+
+import com.pnudev.librarysystem.dto.BookDTO;
+import com.pnudev.librarysystem.dto.RequestBookDTO;
+import com.pnudev.librarysystem.entity.Book;
+import com.pnudev.librarysystem.exception.DeleteFailedException;
+import com.pnudev.librarysystem.exception.EmptyFileException;
+import com.pnudev.librarysystem.exception.FileWrongTypeException;
+import com.pnudev.librarysystem.exception.IOErrorInFileException;
+import com.pnudev.librarysystem.mapper.BookMapper;
+import com.pnudev.librarysystem.repository.BookRepository;
+import com.pnudev.librarysystem.util.SpecificationUtils;
+import jakarta.persistence.EntityNotFoundException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.List;
+import java.util.UUID;
+
+
+@RequiredArgsConstructor
+@Service
+public class BookService {
+
+    @Value("${file.upload-dir}/books")
+    private String uploadDir;
+
+    @Value("${file.image-types}")
+    private List<String> imageTypes;
+
+    private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
+
+
+    public Long createBook(RequestBookDTO requestBookDTO)  {
+        Book book = bookMapper.toEntity(requestBookDTO);
+        Book savedBook = bookRepository.save(book);
+        return savedBook.getId();
+    }
+
+    public BookDTO updateBook(Long id, RequestBookDTO requestBookDTO) {
+        Book book = bookRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Book with id: %d not found".formatted(id)));
```

Comment:
I see a lot of throws with the exact same message format. 
You can extract this as a constant and just pass on the id when needed

Reply:
Done